src/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderReadTests.cs

```diff
@@ -0,0 +1,235 @@
+        {
+            string fst = "<!ENTITY fst 'Sample String'>]>";
+            string dtd = "<!DOCTYPE root[<!ELEMENT root (#PCDATA)>" + fst;
+            string xml = dtd + "<root>&fst;</root>";
+            XmlDocument document = new XmlDocument();
+            document.LoadXml(xml);
+            XmlNodeReader nodeReader = new XmlNodeReader(document);
+            Assert.True(nodeReader.Read()); //DTD Read
+            Assert.True(nodeReader.Read()); //Root Element Read
+            Assert.True(nodeReader.Read()); //EntityReference Read
+            Assert.Equal(ReadState.Interactive, nodeReader.ReadState);
+            Assert.Equal(XmlNodeType.EntityReference, nodeReader.NodeType);
+            nodeReader.ResolveEntity();
+            Assert.True(nodeReader.CanResolveEntity);
+            Assert.Equal("Sample String", nodeReader.ReadString());
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBase64WithEmptyXml()
+        {
+            XmlDocument xmlDoc = new XmlDocument();
+            XmlNodeReader nodeReader = new XmlNodeReader(xmlDoc);
+            Assert.Equal(0, nodeReader.ReadContentAsBase64(null, 0, 0));
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBase64WithSimpleXml()
+        {
+            string xml = "<root attr='val'><child /></root>";
+            XmlDocument document = new XmlDocument();
+            document.LoadXml(xml);
+            XmlNodeReader nodeReader = new XmlNodeReader(document);
+            Assert.True(nodeReader.Read());
+            Assert.True(nodeReader.MoveToAttribute("attr"));
+            Assert.True(nodeReader.CanReadBinaryContent);
+            Assert.Equal(2, nodeReader.ReadContentAsBase64(new byte[33], 10, 10));
+            Assert.Equal(0, nodeReader.ReadContentAsBase64(new byte[33], 10, 10));
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBinHexWithEmptyXml()
+        {
+            XmlDocument xmlDoc = new XmlDocument();
+            XmlNodeReader nodeReader = new XmlNodeReader(xmlDoc);
+            Assert.Equal(0, nodeReader.ReadContentAsBinHex(null, 0, 0));
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBinHexWithSimpleXml()
+        {
+            string xml = "<root attr='ff0000'><child /></root>";
+            XmlDocument document = new XmlDocument();
+            document.LoadXml(xml);
+            XmlNodeReader nodeReader = new XmlNodeReader(document);
+            Assert.True(nodeReader.Read());
+            Assert.True(nodeReader.MoveToAttribute("attr"));
+            Assert.True(nodeReader.CanReadBinaryContent);
+            Assert.Equal(3, nodeReader.ReadContentAsBinHex(new byte[33], 10, 10));
+            Assert.Equal(0, nodeReader.ReadContentAsBinHex(new byte[33], 10, 10));
+        }
+
+        [Fact]
+        public void NodeReaderReadElementContentAsBase64WithEmptyXml()
+        {
+            XmlDocument xmlDoc = new XmlDocument();
```

Comment:
Please use `var` on the left when it's a simple constructor on the right. eg.,
` var xmlDoc = new XmlDocument();`
This makes the code a little quicker to read.

Reply:
Addressed in the new commits